### PR TITLE
Fix slow sound init in piano sample app

### LINKED ISSUE #1696: Slow sound init for piano sample app
The calls to fetchUrl() and prepare() are done sequentially instead of in parallel, so loading the URLs takes 5.3 seconds on my Nexus 5. However, the UI is brought up immediately, so the app appears to be broken when you run it.

### PR DESCRIPTION
**Pull request by jimbeveridge**
Fixes #1696. Also fixes handles leaking on error.

### CODE REVIEW

File: examples/widgets/piano.dart [hunk 1]
```diff
@@ -21,20 +23,30 @@ class Key {
...
     player.ptr.pause();
   }
+
+  Future load(MediaServiceProxy mediaService) async {
+    try {
+      mediaService.ptr.createPlayer(player);
+      UrlResponse response = await fetchUrl(soundUrl);
+      await player.ptr.prepare(response.body);
+    } catch (e) {
+      player.close();
```
Comment (abarth) [inline examples/widgets/piano.dart]: Should we print the exception?

File: examples/widgets/piano.dart [hunk 2]
```diff
@@ -47,47 +59,48 @@ class PianoApp extends StatelessComponent {
...
 
     return new Column(children);
   }
 }
 
-void main() {
-  runApp(new PianoApp());
+main() async {
+  PianoApp app = new PianoApp();
+  // use "await" to make sure the sound files are loaded before we show the ui.
```
Comment (abarth) [inline examples/widgets/piano.dart]: Maybe call `runApp` here with a loading screen?

```dart
runApp(new Center(child: new Text("Loading...")));
```

You can call `runApp` as many times as you like.

**Comment by abarth** [comment]
LGTM with some sort of loading indication.

**Comment by jimbeveridge** [comment]
@abarth Could you take a quick look at the statusBox() function and see if you have any comments?

**Comment by abarth** [comment]
LGTM

**Pull request merged**